### ISSUE TITLE
Services deployed on kubernetes and registered with Eureka are not discoverable  

### ISSUE THREAD

**Issue by focode** [Netflix/eureka#988]
Services deployed on kubernetes are getting registered with Eureka. However the home page URL (Service Endpoint) of the service points to the service POD name which is not an actual host, this results in Unknown host exception when forwarding request via Zuul.

**Comment by wangpenghit2155** [comment]
It is a typical problem. My solution is: you can use NodePort in kubernetes service yaml. And in Dockerfile you can pass "--eureka.instance.nonSecurePort" and "--eureka.instance.ip-address" to your service

**Comment by zhrum** [comment]
@wangpenghit2155 I don't understand what you mean...

**Comment by qiangdavidliu** [comment]
Eureka client allows for customization of several parameters that may be able to help address this issue. I believe the solution outlined by @wangpenghit2155 executes an override of the (auto-discovered) system ip and port that is used for registration. Another potential way to address this is to set the override for the absolute url of the homepage: https://github.com/Netflix/eureka/blob/master/eureka-client/src/main/java/com/netflix/appinfo/EurekaInstanceConfig.java#L293

Closing this issue for now. Please reopen if neither of the above solutions work.

**Comment by dshamanthreddy** [comment]
It is not woking for me

**Comment by cforce** [comment]
Its not working for me too.

As described in the kubernetes manual it should be possible to setup deployments/pods such way, that the (by k8 generated) pod name is registered at the kubernetes DNS (coredns)  so we are able to nslookup by podname additionally to lookup by service name (using the cluster ip)
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pods

We tried to setup the deployment.yaml in different way, but never succeeded so far in the results that is described in the docs, that shall allow a nslookup to be resolved from pod  A for Pod B real IP using
pod-ip-address.my-namespace.pod.cluster.local 

or (what is even more wished)

pod-name.my-namespace.pod.cluster.local 


Like the doc says,  clusterIP: None and we tried with hostname set (or even not set and being generated by kubernetes itself

We verified the test runs using nslookup while ssh on console of Pod A.

With no approach there was ever a DNS resolution possible in Pod A   using B.dev.pod.cluster.local (or just B ) where the podname of B was a generated one which what listed in in k8 pod description and on POD B /etc/hosts  file.
